hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/util/TokenRelListCalculator.java

```diff
@@ -234,21 +234,25 @@ private TokenRelation getInPriorityOrder(
     @NonNull
     private TokenRelPointerUpdateResult updatePointersSurroundingTargetTokenRel(
             @Nullable TokenRelation prevTokenRel, @Nullable final TokenRelation nextTokenRel) {
-        final var prevTokenRelTokenNum = prevTokenRel != null ? prevTokenRel.tokenNumber() : -1;
-        final var nextTokenRelTokenNum = nextTokenRel != null ? nextTokenRel.tokenNumber() : -1;
+        final var prevTokenRelTokenId = prevTokenRel != null
+                ? prevTokenRel.tokenId()
+                : TokenID.newBuilder().tokenNum(-1L).build();
```

Comment:
This should be null

Reply:
Updated. These changes causes the tests to fail. I had to fix some edge cases in this class and TokenDissociateFromAccountHandler to get all the tests to pass. Please check the code change, @Neeharika-Sompalli @mhess-swl .